Fable.Remoting.Client/Proxy.fs

```diff
@@ -51,6 +53,7 @@ module Proxy =
            CustomHeaders          : Map<string, HttpRequestHeaders list>
            Endpoint               : string option
            Headers                : HttpRequestHeaders list
+           TokenCallback          : (unit -> Fable.Import.JS.Promise<string>)
```

Comment:
Since a cached token is only a `Promise.lift` away I figured I could simplify the API even further.

Reply:
Isn't it better to have a `string option` as return instead of a `string`? So we wouldn't need to treat the case of empty string and be more consistent.

Reply:
You're reviewing faster than I can add my comments :) I asked myself the same,  see one of the comments below, so I'll change it to string option then.